checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallConsistencyAnalyzer.java

```diff
@@ -0,0 +1,1602 @@
+          if (AnnotationUtils.areSameByName(
+              anno, "org.checkerframework.checker.calledmethods.qual.CalledMethods")) {
+            cmAnno = anno;
+          }
+        }
+      }
+      if (cmAnno == null) {
+        cmAnno =
+            typeFactory
+                .getAnnotatedType(localVarWithTree.localVar.getElement())
+                .getAnnotationInHierarchy(typeFactory.top);
+      }
+
+      if (calledMethodsSatisfyMustCall(mustCallValue, cmAnno)) {
+        mustCallSatisfied = true;
+        break;
+      }
+    }
+
+    if (!mustCallSatisfied) {
+      LocalVarWithTree firstlocalVarWithTree = fact.iterator().next();
+      if (!reportedMustCallErrors.contains(firstlocalVarWithTree)) {
+        if (!checker.shouldSkipUses(TreeUtils.elementFromTree(firstlocalVarWithTree.tree))) {
+          reportedMustCallErrors.add(firstlocalVarWithTree);
+          checker.reportError(
+              firstlocalVarWithTree.tree,
+              "required.method.not.called",
+              formatMissingMustCallMethods(mustCallValue),
+              firstlocalVarWithTree.localVar.getType().toString(),
+              outOfScopeReason);
+        }
+      }
+    }
+  }
+
+  /**
+   * Increment the -AcountMustCall counter.
+   *
+   * @param node the node being counted, to extract the type
+   */
+  private void incrementNumMustCall(Node node) {
+    if (checker.hasOption(ResourceLeakChecker.COUNT_MUST_CALL)) {
+      TypeMirror type = node.getType();
+      incrementMustCallImpl(type);
+    }
+  }
+
+  /**
+   * Increment the -AcountMustCall counter.
+   *
+   * @param elt the elt being counted, to extract the type
+   */
+  private void incrementNumMustCall(Element elt) {
+    if (checker.hasOption(ResourceLeakChecker.COUNT_MUST_CALL)) {
+      TypeMirror type = elt.asType();
+      incrementMustCallImpl(type);
+    }
+  }
+
+  /**
+   * Shared implementation for the two version of countMustCall. Don't call this directly.
+   *
+   * @param type the type of the object that has a must call obligation
+   */
+  private void incrementMustCallImpl(TypeMirror type) {
+    // only count uses of JDK classes, since that's what we report on in the paper
+    String qualifiedName = TypesUtils.getTypeElement(type).getQualifiedName().toString();
+    if (!qualifiedName.startsWith("java")) {
+      return;
+    }
+    checker.numMustCall++;
+  }
+
+  /**
+   * Do the called methods represented by the {@link CalledMethods} type {@code cmAnno} include all
+   * the methods in {@code mustCallValues}?
+   *
+   * @param mustCallValues the strings representing the must-call obligations
+   * @param cmAnno an annotation from the called-methods type hierarchy
+   * @return true iff cmAnno is a subtype of a called-methods annotation with the same values as
+   *     mustCallValues
+   */
+  private boolean calledMethodsSatisfyMustCall(
+      List<String> mustCallValues, AnnotationMirror cmAnno) {
+    AnnotationMirror cmAnnoForMustCallMethods =
```

Comment:
Why does this code create an annotation rather than doing a subset operation on lists of strings?  Please document the reason.  Creating an annotation seems less efficient and less simple to explain, so the documentation will be helpful.